cmake/modules/configuration_files.cmake

```diff
@@ -27,6 +27,9 @@ include_guard(GLOBAL)
 
 include(extensions)
 
+# Merge in variables from other sources (e.g. sysbuild)
+zephyr_get(FILE_SUFFIX SYSBUILD GLOBAL)
```

Comment:
I do not think it should, for the reason that `zephyr_file_suffix` is completely reusable from modules and out of tree code and has no dependency on files being dts or Kconfig fragment files, it can be used to check for any file type which might be used for any purpose

Reply:
if you want a general function available in addition then that is acceptable, but remove the relative path adjustment and have the caller of the function adjust to abs path before calling this function.

zephyr file can then do the path adjustment, and afterwards use the shared function. 

Else there is a high risk that we end back in the rabbit whole where we have been stuck before.

There is no reason why CMake code should using relative paths, there are plenty of CMake variables to use for identifying where a file should reside, like `${CMAKE_CURRENT_SOURCE_DIR}/<file>`, `${APPLICATION_CONFIG_DIR}/<file>`, etc.

Reply:
Have updated